helix-core/src/selection.rs

```diff
@@ -102,6 +110,21 @@ impl Range {
         self.anchor == self.head
     }
 
+    /// `Some(Direction::Forward)` when head > anchor.
+    /// `Some(Direction::Backward)` when head < anchor.
+    /// None otherwise.
+    #[inline]
+    #[must_use]
+    pub fn direction(&self) -> Option<Direction> {
+        if self.head < self.anchor {
+            Some(Direction::Backward)
+        } else if self.head > self.anchor {
+            Some(Direction::Forward)
+        } else {
+            None
+        }
+    }
```

Comment:
Let's assume a 0-width range is forward facing. No need to handle 0-width ranges specially.

```rust
if self.head < self.anchor { Direction::Backward } else { Direction::Forward }
```

Reply:
Done